### PR TITLE
sdxl: fix output_path argument issue and reformat

### PR DESCRIPTION
**Pull request by chenyang23333**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 1]
```diff
@@ -279,8 +279,7 @@ def convert_openai_text_enc_state_dict(text_enc_dict):
 
 
 def convert_weight(pth, msname):
-    sd = torch.load(pth)
-    key_torch = list(sd["state_dict"].keys())
+    key_torch = list(pth["state_dict"].keys())
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:282]: `pth` -> `state_dict`

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 2]
```diff
@@ -303,16 +302,16 @@ def convert_weight(pth, msname):
     ms_key = []
     for i in range(len(key_torch)):
         kt, kms = key_torch[i], key_ms[i]
-        newckpt.append({"name": kms, "data": Tensor(sd["state_dict"][kt].numpy(), ms.float32)})
+        newckpt.append({"name": kms, "data": Tensor(pth["state_dict"][kt].numpy(), ms.float32)})
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:305]: 同上

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 3]
```diff
@@ -341,75 +340,78 @@ def compare_missing_key(key_list):
...
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("--model_path", default=None, type=str, required=True, help="Path to the model to convert.")
-    parser.add_argument("--checkpoint_path", default=None, type=str, required=True, help="Path to the output model.")
+    parser.add_argument("--output_path", default=None, type=str, required=True, help="Path to the output model.")
     parser.add_argument("--half", action="store_true", help="Save weights in half precision.")
     parser.add_argument("--use_safetensors", action="store_true", help="Save weights use safetensors, default is ckpt.")
     parser.add_argument(
-        "--unet_name", default="diffusion_pytorch_model.fp16.safetensors", type=str, help="Name to the unet model."
+        "--unet_path", default="diffusion_pytorch_model.fp16.safetensors", type=str, help="Path to the unet model."
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:347]: default="diffusion_pytorch_model.fp16.safetensors" 
=>
default="./unet/diffusion_pytorch_model.fp16.safetensors"

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 4]
```diff
@@ -341,75 +340,78 @@ def compare_missing_key(key_list):
...
+        "--vae_path", default="diffusion_pytorch_model.fp16.safetensors", type=str, help="Path to the vae model."
     )
     parser.add_argument(
-        "--text_encoder_name", default="model.fp16.safetensors", type=str, help="Name to the text_encoder model."
+        "--text_encoder_path", default="model.fp16.safetensors", type=str, help="Path to the text_encoder model."
     )
     parser.add_argument(
-        "--text_encoder_2_name", default="model.fp16.safetensors", type=str, help="Name to the text_encoder_2 model."
+        "--text_encoder_2_path", default="model.fp16.safetensors", type=str, help="Path to the text_encoder_2 model."
+    )
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:357]: 同上

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 5]
```diff
@@ -341,75 +340,78 @@ def compare_missing_key(key_list):
...
+    parser.add_argument("--sdxl_base_ckpt", default=None, type=str, help="Path to the sd_xl_base model.")
+    parser.add_argument(
+        "--save_type",
+        default=2,
+        type=int,
+        help="Output weight type: 0 for Torch SafeTensors, 1 for Torch .pth, 2 for MindSpore .ckpt",
     )
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:364]: `--save_type` 更换为 `--save_mindspore`, `--save_torch`, `--save_safetensor`

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 6]
```diff
@@ -341,75 +340,78 @@ def compare_missing_key(key_list):
...
+    # unet_path = osp.join(args.model_path, "unet", args.unet_path)
+    # vae_path = osp.join(args.model_path, "vae", args.vae_path)
+    # text_enc_path = osp.join(args.model_path, "text_encoder", args.text_encoder_path)
+    # text_enc_2_path = osp.join(args.model_path, "text_encoder_2", args.text_encoder_2_path)
+    #
+    # # Load models from safetensors if it exists, if it doesn't pytorch
+    # unet_bin_path = osp.join(args.model_path, "unet", "diffusion_pytorch_model.bin")
+    # vae_bin_path = osp.join(args.model_path, "vae", "diffusion_pytorch_model.bin")
+    # text_enc_bin_path = osp.join(args.model_path, "text_encoder", "pytorch_model.bin")
+    # text_enc_2_bin_path = osp.join(args.model_path, "text_encoder_2", "pytorch_model.bin")
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:382]: delete

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 7]
```diff
@@ -341,75 +340,78 @@ def compare_missing_key(key_list):
...
-
-    elif osp.exists(unet_bin_path):
-        unet_state_dict = torch.load(unet_bin_path, map_location="cpu")
-        print("load unet from unet_bin_path： ", unet_bin_path)
-
-    if osp.exists(vae_path):
-        vae_state_dict = load_file(vae_path, device="cpu")
-        print("load vae from vae_path ", vae_path)
+    if osp.exists(args.unet_path):
+        if args.unet_path.endswith("bin"):
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:389]: `bin` -> `.bin`

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 8]
```diff
@@ -341,75 +340,78 @@ def compare_missing_key(key_list):
...
+            unet_state_dict = torch.load(args.unet_bin_path, map_location="cpu")
+        else:
+            unet_state_dict = load_file(args.unet_path, device="cpu")
+        print("load unet from unet_path: ", args.unet_path)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:393]: `if osp.exists(args.unet_path):` 增加else分支和打印

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 9]
```diff
@@ -431,38 +433,44 @@ def compare_missing_key(key_list):
...
+    elif args.save_type == 1:
         state_dict = {"state_dict": state_dict}
-        torch.save(state_dict, "torch_part.ckpt")
-
-    # Convert the torch ckpt to mindspore ckpt and return mindspore key list
-    key_list = convert_weight("torch_part.ckpt", "mindspore_part.ckpt")
+        torch.save(state_dict, args.output_path)
+    elif args.save_type == 2:
+        # Convert the torch ckpt to mindspore ckpt and return mindspore key list
+        key_list = convert_weight(state_dict, args.output_path)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:444]: 同上

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 10]
```diff
@@ -431,38 +433,44 @@ def compare_missing_key(key_list):
...
-            missing_key_list = compare_missing_key(key_list)
-            print("The first 20 missing parameters are: ", str(missing_key_list[:20]))
-        if not args.sdxl_base_ckpt:
+            with open("mindspore_key_base.yaml", "r") as file:
+                line_count = len(file.readlines())
+        else:
+            line_count = 2514
+
+        # If you have obtained all the keys, you do not need to run the insertion operation
+        if len(key_list) == line_count or not args.sdxl_base_ckpt:
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:453]: 1、`len(key_list) == line_count` 只能说明数量一样，可以增加必要的key match的比较
2、两个条件判断分开两个分支

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 11]
```diff
@@ -431,38 +433,44 @@ def compare_missing_key(key_list):
...
+                print(
+                    "you have added sdxl_base_ckpt argument,so it will run merge operation(Integrate the retrieved "
+                    "MindSpore checkpoint into the sdxl base model checkpoint)"
+                )
+                merge_weight(args.output_path, args.sdxl_base_ckpt)
+
+        if len(key_list) > line_count:
+            raise ValueError(
+                "The number of keys is greater than mindspore sd xl base checkpoint. Insertion not allowed."
             )
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:476]: 把多出来的key和数量打印出来

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 12]
```diff
@@ -322,11 +321,11 @@ def merge_weight(partckpt, allckpt):
...
-    ms.save_checkpoint(newckpt, args.checkpoint_path)
+    ms.save_checkpoint(newckpt, args.output_path)
     print("insert Stable Diffusion checkpoint(mindspore) to sd_xl_base_1.0_ms.ckpt success!")
 
 
-def compare_missing_key(key_list):
+def compare_missing_key(key_list, reverse=False):
     # get the sdxl base ckpt's key
     base_key = []
     with open("mindspore_key_base.yaml", "r", encoding="utf-8") as file:
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:331]: 入参增加 mindspore_key.yaml

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 13]
```diff
@@ -335,134 +334,185 @@ def compare_missing_key(key_list):
...
         else:
-            print(
-                "you have added sdxl_base_ckpt argument,so it will run merge operation(merge mindspore_part.ckpt to sdxl_base_ckpt)"
+            line_count = 2514
+
+        if not args.sdxl_base_ckpt:
+            print("you did not supply the 'sdxl_base_ckpt' argument; hence, the insertion operation was not executed.")
+
+        # If you have obtained all the keys, you do not need to run the insertion operation
+        elif len(key_list) == line_count and len(compare_missing_key(key_list)) == 0:
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:491]: compare_missing_key 需要在 `if osp.exists("mindspore_key_base.yaml"):` 的判断分支内

File: examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py [hunk 14]
```diff
@@ -335,134 +334,185 @@ def compare_missing_key(key_list):
...
+        if len(key_list) > line_count:
+            print("The MindSpore checkpoint (.ckpt) file that we have obtained contains ", str(len(key_list)), "keys.")
+            print(str(len(key_list) - line_count), " more parameters than sdxl base ckpt")
+            if osp.exists("mindspore_key_base.yaml"):
+                more_key_list = compare_missing_key(key_list, True)
+                print("The first 20 more parameters are: ", str(more_key_list[:20]))
+
+            raise ValueError(
+                "The number of keys is greater than mindspore sd xl base checkpoint. Insertion not allowed."
             )
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/model_conversion/convert_diffusers_to_mindone_sdxl.py:518]: miss/more 可能同时出现 不要用len的长度做区分，直接compare一次 把miss/more的数量和参数名(前20个)打印出来

**Review by SamitHuang — APPROVED** [review]

**Review by zhanghuiyao — APPROVED** [review]

**Pull request merged**